internal/irc/handler.go

```diff
@@ -492,6 +502,20 @@ func (h *Handler) handleNickServ(msg ircmsg.Message) {
 	}
 }
 
+// botModeSupported checks if IRCv3 Bot Mode is supported by the server
+// See https://ircv3.net/specs/extensions/bot-mode
+func (h *Handler) botModeSupported() bool {
+	h.botModeChar = h.client.ISupport()["BOT"]
```

Comment:
Could this fail if it's named something else? Could be safer to check the map first for the result and only set if it's there.

Reply:
[`Connection.ISupport()`](https://github.com/ergochat/irc-go/blob/5474a63ef6b9e03232499f9d8b6fb646a6a2d7f5/ircevent/irc.go#L531C29-L531C29) returns a `map[string]string`. If `"BOT"` is not found in the map, it will simply return an empty string (the default string value), so it should already be safe.  
A similar check is done in the [irc-go samples](https://github.com/ergochat/irc-go/blob/5474a63ef6b9e03232499f9d8b6fb646a6a2d7f5/ircevent/examples/simple.go#L42), for example.